airflow/api/client/api_client.py

```diff
@@ -16,14 +16,17 @@
 # specific language governing permissions and limitations
 # under the License.
 """Client for all the API clients."""
+import requests
 
 
 class Client:
     """Base API client for all API clients."""
 
-    def __init__(self, api_base_url, auth):
+    def __init__(self, api_base_url, auth=None, session=None):
```

Comment:
Would you mind adding type hints? There's ongoing effort to increase coverage #9783

Reply:
This entire module has no types. I'm afraid that if I add annotations in one place, I will have to make a lot of additional changes.
I can see that the IDE already shows me various errors, and then it can get even worse.
<img width="899" alt="Screenshot 2020-07-22 at 10 56 42" src="https://user-images.githubusercontent.com/12058428/88156750-23c3df80-cc0a-11ea-8630-342e95d90f87.png">
After that, we should retire this API client with Airflow 2.0 and start using the OpenAPI based client.